Fix in-game voting

Partmedia opened the pull request:
## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->
A regression introduced after the UI refactor caused the vote-casting window to not appear when the client was in-game.

*This fix was sponsored by Mining Station 14.*

**Steps To Reproduce**
Enter the game as a ghost or player, call a vote. You hear the "ding" but no vote window shows up.

**Media**
![2022-12-08_11-05](https://user-images.githubusercontent.com/3229565/206546771-a074dacb-38c8-46a9-a71d-2926d5d0d956.png)

- [X] I have added screenshots/videos to this PR showcasing its changes ingame
- [ ] This PR does not require an ingame showcase

**Changelog**
:cl: notafet
- fix: Votes can once again be cast in-game

moonheart08 reviewed (changes requested):
This fix is an ugly hack.

Add a new UI widget to the different states/screens that votes should be put in, and set the popup container to it when switching between states.

Partmedia commented:
While I agree that your suggested fix would be better, I also think that this fix is better than no fix. Besides, there is precedent for this type of fix (see 85866ea).

mirrorcult commented:
this fix isn't ideal but it's certainly not ugly and it's better than nothing, imo just add a comment that this should be removed in favor of a proper popup eventually

mirrorcult commented:
also the restart vote cvar should be reenabled

mirrorcult commented:
also the restart vote cvar should be reenabled

moonheart08 commented:
I don't really want to budge on this considering adding UI elements to each state for the vote container isn't very hard.

Partmedia commented:
I've added a comment about the "hack" and re-enabled vote restart by default, as requested.

We're all volunteers, and I'm not presently interested in adding and testing per-state vote containers. I investigated and fixed making votes functional again, and thought upstream might find the fix useful.
